### PR TITLE
Build examples.py

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: Tools/autotest/pysim/util.py [hunk 1]
```diff
@@ -64,25 +64,45 @@ def deltree(path):
...
+
+    # first configure
+    waf_configure(board, j=j, debug=debug)
+
+    # then clean
+    waf_clean()
+
     # then build
-    cmd_make = [relwaf(), "build", "--target", build_target]
+    cmd_make = [relwaf(), 'build', "--target", build_target]
```
Comment (lucasdemarchi) [inline Tools/autotest/pysim/util.py]: Why did you change it for build but not for --target?

Comment (peterbarker) [inline Tools/autotest/pysim/util.py]: Huh.  I didn't mean to change it at all!  The refactoring in there changed shape a couple of times.

Is there any preference in Python for one or the other?  There was an extremely slight performance advantage in Perl in using the non-interpolating '' vs the interpolating ""

File: Tools/autotest/pysim/util.py [hunk 2]
```diff
@@ -64,25 +64,45 @@ def deltree(path):
...
     run_cmd(cmd_make, dir=topdir(), checkfail=True, show=True)
```
Comment (lucasdemarchi) [inline Tools/autotest/pysim/util.py]: No, they are equivalent. I suggest running pep8 and check if it recommends anything

File: Tools/autotest/pysim/util.py (hunk 1 above)
Comment (lucasdemarchi) [inline Tools/autotest/pysim/util.py]: No, they are equivalent. I suggest running pep8 and check if it recommends anything

Comment (lucasdemarchi) [inline Tools/autotest/pysim/util.py]: I removed this change while rebasing.

**Comment by lucasdemarchi** [comment]
Applied, thanks.

**Comment by OXINARF** [comment]
@peterbarker I just noticed that you are passing the jobs flag to the configure step, but I'm 99% sure that that doesn't do anything, you need to send the jobs to the build step.

**Comment by peterbarker** [comment]
On Sat, 4 Jun 2016, Francisco Ferreira wrote:

> @peterbarker I just noticed that you are passing the jobs flag to the configure step, but I'm 99% sure that that doesn't do anything, you need to send the
> jobs to the build step.

Done, thanks.
